### ISSUE TITLE
GELF Logging policy

### ISSUE THREAD

**Issue by Frintrop** [gravitee-io/issues#445]
We would write a gelf logging policy to log directly to graylog,logstash.

a) depending external jar will be deployed within the policy jar itself ?
b) logging could be async to request chain itself. could we write straight forward a logging policy like yours stdout logging policy or do we build our async processing within the policy itself ?!

**Comment by brasseld** [comment]
Hi @Frintrop,

Can you tell us what do you want to log exactly ?
Request / response content (headers, body, ...) ?

**Comment by Frintrop** [comment]
yes. all. to an external log server. ( instead using runscope in front )

**Comment by brasseld** [comment]
Get it !

So:
a) External jars will be deployed within the policy **zip** itself (because a gravitee.io plugin is a zip file). Please have a look to a policy example under ${GRAVITEE_HOME}/plugins
b) Logging **must** be done asynchronously.

But IMHO, this should not be part of the policy features. `Logging` policy has been developed and provided just to provide a way to have a quick look to the actual request / response, and is just use for `debugging` purpose.

To provide you with a complete answer, I need more information from you. Currently, you want to log request / response into an external logging system. But do you want to log the incoming request or the outbound request? In the same way, you want to log the response provided by your API or the one returned to the consumer in front of the api gateway ?

**Comment by Frintrop** [comment]
I want to log  incoming request and responses from clients to the api gateway...
